lib/pac55xx/ccs.c

```diff
@@ -0,0 +1,271 @@
+/**
+ * ccs.c
+ * @brief <b>PAC55xxxx CCS Driver</b>
+ * @author @htmlonly &copy; @endhtmlonly 2020 Kevin Stefanik <kevin@allocor.tech>
+ * @date March 7, 2020
+ *
+ * This library supports the CCS module in the PAC55xx SoC from Qorvo.
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <libopencm3/pac55xx/ccs.h>
+#include <libopencm3/pac55xx/memorymap.h>
+#include <libopencm3/pac55xx/memctl.h>
+#include <libopencm3/cm3/assert.h>
+
+/** Global clock tree structure that is configured by ccs_configure_clocks */
+uint32_t ccs_extclk_frequency = CCS_ROSC_FREQ;
```

Comment:
Consider making these static (global write access to these seems bad)

Reply:
they are now static